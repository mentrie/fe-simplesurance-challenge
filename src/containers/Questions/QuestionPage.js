import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../../components/TextInput';

const QuestionPage = ({value, onChange, onNextStep, sendCustomerResponse, error}) => {
  const onClick = value.next? onNextStep: sendCustomerResponse
  return (
    <div className="questions">
      <section className="questions__header">
        <h3 className="header">Welcome To Customer Complaint Portal</h3>
        <p className="instruction">(Please, take your time to answer all the question to help us process your complaint).</p>
      </section>
      <TextInput 
        type={value.type}
        value={value}
        onChange={onChange}
        question={value.text}
        error={error}
      />
      <button className="question__next__btn" onClick={onClick}>Next</button>
    </div>
  );
}

QuestionPage.propTypes = {
 value: PropTypes.shape({
   next: PropTypes.any(['object', 'string']),
   text: PropTypes.string.isRequired,
   reply: PropTypes.string,
   type: PropTypes.string
 }),
 onChange: PropTypes.func.isRequired,
 sendCustomerResponse: PropTypes.func,
 error: PropTypes.string
}

export default QuestionPage