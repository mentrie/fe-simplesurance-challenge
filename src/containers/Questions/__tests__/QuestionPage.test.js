import React from 'react'
import QuestionPage from '../QuestionPage';

test('should render page correctly', () => {
  const props = {
    value: {
      type: '',
      next: null,
      text: 'Testing'
    }
  }

  const component = shallow(<QuestionPage {...props} />);
  expect(component).toMatchSnapshot();
});

test('should call on next question when button is clicked', () => {
  const props = {
    value: {
      type: '',
      next: '34',
      text: 'Testing'
    },
    onNextStep: jest.fn()
  }

  const component = mount(<QuestionPage {...props} />);
  component.find('button').simulate('click');
  expect(component.props().onNextStep).toHaveBeenCalled();
});