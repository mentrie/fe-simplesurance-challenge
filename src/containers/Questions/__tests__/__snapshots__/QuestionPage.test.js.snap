// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page correctly 1`] = `
<div
  className="questions"
>
  <section
    className="questions__header"
  >
    <h3
      className="header"
    >
      Welcome To Customer Complaint Portal
    </h3>
    <p
      className="instruction"
    >
      (Please, take your time to answer all the question to help us process your complaint).
    </p>
  </section>
  <TextInput
    question="Testing"
    type=""
    value={
      Object {
        "next": null,
        "text": "Testing",
        "type": "",
      }
    }
  />
  <button
    className="question__next__btn"
  >
    Next
  </button>
</div>
`;
